File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    answer = 0
    num_from = 0
    
    for i in range(len(t)-len(p)+1):
        num_from = i+len(p)
        if int(t[i:num_from]) <= int(p):
            answer = answer + 1
    
    return answer

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        int pLen = p.length();
        String tCopy = t;
        long num = 0;
        
        for(int i=0; i<=t.length()-pLen; i++) {
            tCopy = t;
            num = Long.parseLong(tCopy.substring(i, i+pLen));
            if(num <= Long.parseLong(p))
                answer++;
        }
        return answer;
    }
}